### PR TITLE
[Console] Fix signal management 

### PR DESCRIPTION
**Pull request by chalasr**
| Q             | A
| ------------- | ---
| Branch?       | 5.2
| Bug fix?      | yes
| New feature?  | no
| Deprecations? | no
| Tickets       | -
| License       | MIT
| Doc PR        | -

Missed in #38589

### CODE REVIEW

**Comment by lyrixx** [comment]
There were other issues

https://github.com/symfony/symfony/blob/cbfa540b02394ce103f6235b58296cfda22dea07/src/Symfony/Component/Console/Application.php#L289

https://github.com/symfony/symfony/blob/cbfa540b02394ce103f6235b58296cfda22dea07/src/Symfony/Component/Console/Application.php#L961

And this check is wrong too. I mean the error message does not reflect what is going. https://github.com/symfony/symfony/blob/cbfa540b02394ce103f6235b58296cfda22dea07/src/Symfony/Component/Console/Application.php#L957-L958

File: src/Symfony/Component/Console/Application.php [hunk 1]
```diff
@@ -108,7 +108,7 @@ public function setCommandLoader(CommandLoaderInterface $commandLoader)
         $this->commandLoader = $commandLoader;
     }
 
-    public function getSignalRegistry(): SignalRegistry
+    public function getSignalRegistry(): ?SignalRegistry
```
Comment (lyrixx) [inline src/Symfony/Component/Console/Application.php]: I don't like this approach. It's a bit poor in term of DX.
People will have to check if the SignalRegistry is defined before using it.
And if it's not defined, they don't know why it's not defined. They have to read the Symfony Code to understand what's it's going.

Instead, I would revert this change and I would throw an exception, something like this one:

https://github.com/symfony/symfony/blob/cbfa540b02394ce103f6235b58296cfda22dea07/src/Symfony/Component/Console/Application.php#L958

Comment (chalasr) [reply src/Symfony/Component/Console/Application.php]: 👍  should be better now

**Review by lyrixx — APPROVED** [review]
👍🏼 
Thank you Robin for taking time of fixing this :) 
I'm gonna change the title of your PR to get a "nice" git history

**Comment by chalasr** [comment]
> I'm gonna change the title of your PR to get a "nice" git history

Broader, but not wrong :) Thanks!

**Comment by lyrixx** [comment]
You were fast on this one, thanks Robin.

**Pull request merged**